### PR TITLE
feat: report cwctl events to hub

### PR DESCRIPTION
**Pull request by vishnu-narayanan**
# Pull Request Template

## Description

Fixes https://linear.app/chatwoot/issue/CW-2481/report-cwctl-usage-to-hub

- [x] Add rake task to get installation_identifier

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- Tested locally
- Tested  remotely on an existing Chatwoot installation
- Tested on a fresh installation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: deployment/setup_20.04.sh [hunk 1]
```diff
@@ -848,6 +849,68 @@ function webserver() {
...
+  local installation_identifier
+
+  sudo -i -u chatwoot << "EOF"
+  cd chatwoot
+  installation_identifier=$(bundle exec rake instance_id:get_installation_identifier)
+EOF
+
+  # Check if the installation identifier is defined
+  if [ -z "$installation_identifier" ]; then
+    echo "Installation identifier is not defined. Unable to report event."
```
Comment (pranavrajs) [inline deployment/setup_20.04.sh:907]: This is a debug log, right? Should we show this to the user?

File: deployment/setup_20.04.sh [hunk 2]
```diff
@@ -848,6 +849,68 @@ function webserver() {
...
+##############################################################################
+function report_event() {
+  local event_name="$1"
+  local event_data="$2"
+
+  CHATWOOT_HUB_URL="https://hub.2.chatwoot.com/events"
+  
+  # Check if the CHATWOOT_HUB_URL is defined
+  if [ -z "$CHATWOOT_HUB_URL" ]; then
+    echo "CHATWOOT_HUB_URL is not defined. Unable to report event."
```
Comment (pranavrajs) [inline deployment/setup_20.04.sh:872]: Same comment as below. This seems like a debug log, we don't need to show this to the user ,right?

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Minor comments, otherwise looks good.

File: deployment/setup_20.04.sh (hunk 1 above)
Comment (vishnu-narayanan) [reply deployment/setup_20.04.sh]: Yes, removed.

File: deployment/setup_20.04.sh (hunk 2 above)
Comment (vishnu-narayanan) [reply deployment/setup_20.04.sh]: Yes, removed.

**Pull request merged**